src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx

```diff
@@ -40,101 +47,181 @@ export const PupilViewsProgrammeListing = ({
     programmes,
     factorPrefix: "tier",
   });
-
   const examboards = getAvailableProgrammeFactor({
     programmes,
     factorPrefix: "examboard",
   }) as ExamboardData[];
 
-  const examboardData =
+  const [chosenExamboard, setChosenExamboard] = useState<ExamboardData | null>(
     examboardSlug && examboards.length >= 1
       ? getExamboardData({ examboardSlug, availableExamboards: examboards })
-      : null;
-
-  const [chosenExamboard, setChosenExamboard] = useState<ExamboardData | null>(
-    examboardData,
+      : null,
   );
 
+  if (!programmes[0]) {
+    throw new Error("No programme data available");
+  }
+  const firstProgrammeFields = programmes[0].programmeFields;
+  const { phaseSlug } = firstProgrammeFields;
+
+  if (phaseSlug === "foundation") {
+    throw new Error("Foundation phase is not supported");
+  }
+
+  // const examboardData =
+
   const backlink = resolveOakHref({
     page: "pupil-subject-index",
     yearSlug,
   });
 
+  const subjectSlug = programmes[0]?.programmeFields.subjectSlug;
+  const subjectDescription = programmes[0]?.programmeFields.subject;
+  const yearDescriptions = programmes[0]?.programmeFields.yearDescription;
+
+  const topNavSlot = () => {
```

Comment:
all the changeable objects are handled by these switch functions and theyre called with the jsx